PGM/src/main/java/tc/oc/pgm/picker/PickerMatchModule.java

```diff
@@ -288,29 +295,43 @@ public void closeMonitoredInventory(final InventoryCloseEvent event) {
     }
 
     @EventHandler
-    public void rightClickIcon(final ObserverInteractEvent event) {
-        if(event.getClickType() != ClickType.RIGHT) return;
-
+    public void handleClickIcon(final ObserverInteractEvent event) {
         MatchPlayer player = event.getPlayer();
-        if(!canUse(player)) return;
 
         ItemStack hand = event.getClickedItem();
         if(ItemUtils.isNothing(hand)) return;
 
         String displayName = hand.getItemMeta().getDisplayName();
         if(displayName == null) return;
 
-        if(hand.getType() == Button.JOIN.material) {
-            event.setCancelled(true);
-            if(canOpenWindow(player)) {
-                showWindow(player);
-            } else {
-                // If there is nothing to pick, just join immediately
-                jmm.requestJoin(player, JoinRequest.user());
-            }
-        } else if(hand.getType() == Button.LEAVE.material) {
```

Comment:
What about the leave button? Does it still work?

Reply:
i simply refactored it and commented it out below